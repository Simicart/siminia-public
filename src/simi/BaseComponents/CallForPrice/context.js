import React, { createContext, useContext } from 'react';
import { useCallForPrice } from './talons/useCallForPrice';

const CallForPriceContext = createContext();
const { Provider } = CallForPriceContext;

const CallForPriceProvider = props => {
    const { children } = props;

    const talonProps = useCallForPrice();
    const { error } = talonProps;

    if (error) {
        if (process.env.NODE_ENV !== 'production') {
            console.error(error);
        }
    }

    const contextValue = {
        ...talonProps
    };

    return <Provider value={contextValue}>{children}</Provider>;
};

export default CallForPriceProvider;

export const useCallForPriceContext = () => useContext(CallForPriceContext);
