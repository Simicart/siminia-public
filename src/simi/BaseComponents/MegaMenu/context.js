import React, { createContext, useContext } from 'react';
import { useMegaMenu } from './talons/useMegaMenu';

const AmMegaMenuContext = createContext();
const { Provider } = AmMegaMenuContext;

const AmMegaMenuProvider = props => {
    const { children } = props;

    const talonProps = useMegaMenu();
    const { error } = talonProps;

    if (error) {
        if (process.env.NODE_ENV !== 'production') {
            console.error(error);
        }
    }

    const contextValue = {
        ...talonProps
    };

    return <Provider value={contextValue}>{children}</Provider>;
};

export default AmMegaMenuProvider;

export const useAmMegaMenuContext = () => useContext(AmMegaMenuContext);
