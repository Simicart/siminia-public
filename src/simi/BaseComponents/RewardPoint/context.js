// import React, { createContext, useContext } from 'react';
// import { useRewardPoint } from './talons/useRewardPoint';

// const BssRewardPointContext = createContext();
// const { Provider } = BssRewardPointContext;

// const BssRewardPointProvider = props => {
//     const { children } = props;

//     const talonProps = useRewardPoint();
//     const { error } = talonProps;

//     if (error) {
//         if (process.env.NODE_ENV !== 'production') {
//             console.error(error);
//         }
//     }

//     const contextValue = {
//         ...talonProps
//     };

//     return <Provider value={contextValue}>{children}</Provider>;
// };

// export default BssRewardPointProvider;

// export const useAmMegaMenuContext = () => useContext(BssRewardPointContext);
