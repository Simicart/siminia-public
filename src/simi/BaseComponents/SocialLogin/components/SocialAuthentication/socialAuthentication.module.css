.root {
  padding-top: 1rem;
  position: relative;
  background: rgb(var(--venia-global-color-gray));
  margin: 3rem 0 0 0;
}

.root + button {
  margin-top: 1rem;
}

@media (min-width: 641px) {
  .root:not(.checkout_cart) {
    margin: 3rem 1.5rem 1rem 1.5rem;
  }

  .root + button {
    margin-top: 0;
  }
}

.root_popup {
  composes: root;
  margin: 1rem 1.5rem;
  grid-area: socialLogin;
}

.title {
  padding: 0.5rem 0.5rem 0 0.5rem;
  text-align: center;
}

.separator {
  position: absolute;
  left: 0;
  right: 0;
  border-bottom: 1px solid rgb(var(--venia-global-color-border));
  top: -2rem;
}

.separatorIcon {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 2rem;
  height: 2rem;
  border: 1px solid rgb(var(--venia-global-color-border));
  border-radius: 1rem;
  background: #fff;
  transform: translate(-50%, -50%);
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: 0.7rem;
  text-transform: uppercase;
  color: rgb(var(--venia-global-color-gray-600));
}

.left,
.right {
  margin-top: 4.5rem;
}

.right .separator,
.left .separator {
  right: initial;
  left: -2rem;
  top: 0;
  bottom: 0;
  border-bottom: 0;
  border-left: 1px solid rgb(var(--venia-global-color-border));
}

.left .separator {
  left: initial;
  right: -2rem;
}

.top .separator,
.bottom .separator {
  top: initial;
  left: 0;
  right: 0;
  bottom: -2rem;
  border-bottom: 1px solid rgb(var(--venia-global-color-border));
}

.bottom .separator {
  bottom: initial;
  top: -2rem;
}
