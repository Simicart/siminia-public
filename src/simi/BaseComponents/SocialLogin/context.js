import React, { createContext, useContext } from 'react';
import { useSocialLogin } from './talons/useSocialLogin';
import { useSignInEvent } from './talons/useSignInEvent';

const AmSocialLoginContext = createContext();
const { Provider } = AmSocialLoginContext;

const AmAdvancedReviewsProvider = props => {
  const { children } = props;

  const talonProps = useSocialLogin();
  const { error, needRedirect, redirectUrl, handleErrors } = talonProps;
  const { isBusy } = useSignInEvent({
    needRedirect,
    redirectUrl,
    handleErrors
  });

  if (error) {
    if (process.env.NODE_ENV !== 'production') {
      console.error(error);
    }

    console.log('Check if Amasty modules has been installed!');
  }

  const contextValue = {
    ...talonProps,
    isBusy
  };

  return <Provider value={contextValue}>{children}</Provider>;
};

export default AmAdvancedReviewsProvider;

export const useSocialLoginContext = () => useContext(AmSocialLoginContext);
