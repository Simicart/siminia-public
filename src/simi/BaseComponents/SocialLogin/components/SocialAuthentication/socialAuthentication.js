import React from 'react';
import { mergeClasses } from '@magento/venia-ui/lib/classify';
import { bool, shape, string } from 'prop-types';
import defaultClasses from './socialAuthentication.module.css';
import SocialButtons from '../SocialButtons';
import { useSocialLoginContext } from '../../context';
import { FormattedMessage } from 'react-intl';
import { useUserContext } from '@magento/peregrine/lib/context/user';

const SocialAuthentication = props => {
  const { mode, isPopup, showCreateAccount } = props;
  const {
    buttons,
    isEnabled,
    buttonPosition,
    enabledModes
  } = useSocialLoginContext();

  const [{ isSignedIn }] = useUserContext();

  if (!isEnabled || !enabledModes.includes(mode) || isSignedIn) {
    return null;
  }

  const classes = mergeClasses(defaultClasses, props.classes);

  const separator =
    mode === 'popup' ? (
      <div className={classes.separator}>
        <span className={classes.separatorIcon}>
          <FormattedMessage
            id={'amSocialLogin.separator'}
            defaultMessage={'or'}
          />
        </span>
      </div>
    ) : null;

  const rootClass = isPopup
    ? `${classes.root_popup} ${classes[buttonPosition]}`
    : `${classes.root} ${classes[mode]}`;

  return (
    <div className={rootClass}>
      {separator}
      <div className={classes.title}>
        <FormattedMessage
          id={'amSocialLogin.titleText'}
          defaultMessage={'Login with Your Social Profile'}
        />
      </div>
      <SocialButtons
        buttons={buttons}
        showCreateAccount={showCreateAccount}
        mode={mode}
      />
    </div>
  );
};

SocialAuthentication.propTypes = {
  classes: shape({
    root: string
  }),
  mode: string,
  isPopup: bool
};

export default SocialAuthentication;
