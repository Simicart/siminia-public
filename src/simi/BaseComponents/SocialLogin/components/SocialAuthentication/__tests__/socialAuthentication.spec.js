import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import SocialAuthentication from '../socialAuthentication';
import { useSocialLoginContext } from '../../../context';

const defaultTalons = {
  storeConfig: null,
  buttons: [
    {
      label: 'Facebook',
      type: 'facebook',
      url:
        'https://ce241-pwa--atakarski.ap74.corp.amdev.by/amsociallogin/social/login/?type=facebook'
    },
    {
      label: 'Instagram',
      type: 'instagram',
      url:
        'https://ce241-pwa--atakarski.ap74.corp.amdev.by/amsociallogin/social/login/?type=instagram'
    }
  ],
  isEnabled: true,
  buttonShape: 'round',
  buttonPosition: 'top',
  needRedirect: false,
  errors: null,
  handleErrors: jest.fn(),
  enabledModes: 'popup,checkout_cart'
};

jest.mock('../../../context', () => {
  const useSocialLoginContext = jest.fn(() => {});

  return { useSocialLoginContext };
});
jest.mock('@magento/peregrine/lib/context/user', () => {
  const userState = {
    isGettingDetails: false,
    getDetailsError: null
  };
  const userApi = {
    getUserDetails: jest.fn(),
    setToken: jest.fn(),
    signIn: jest.fn()
  };
  const useUserContext = jest.fn(() => [userState, userApi]);

  return { useUserContext };
});
jest.mock('react-intl', () => ({
  FormattedMessage: props => (
    <div componentName="Formatted Message Component" {...props} />
  ),
  useIntl: jest.fn().mockReturnValue({
    formatMessage: jest
      .fn()
      .mockImplementation(options => options.defaultMessage)
  })
}));

test('render empty', () => {
  useSocialLoginContext.mockReturnValue({ isEnabled: false, buttons: null });

  const tree = createTestInstance(<SocialAuthentication />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('render disabled mode', () => {
  useSocialLoginContext.mockReturnValue(defaultTalons);

  const tree = createTestInstance(<SocialAuthentication mode="register" />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('render in popup', () => {
  useSocialLoginContext.mockReturnValue(defaultTalons);

  const tree = createTestInstance(<SocialAuthentication mode="popup" />);
  expect(tree.toJSON()).toMatchSnapshot();
});
