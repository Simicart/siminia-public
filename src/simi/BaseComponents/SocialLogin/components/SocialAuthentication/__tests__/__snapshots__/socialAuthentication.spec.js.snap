// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render disabled mode 1`] = `null`;

exports[`render empty 1`] = `null`;

exports[`render in popup 1`] = `
<div
  className="undefined undefined"
>
  <div>
    <span>
      <div
        componentName="Formatted Message Component"
        defaultMessage="or"
        id="amSocialLogin.separator"
      />
    </span>
  </div>
  <div>
    <div
      componentName="Formatted Message Component"
      defaultMessage="Login with Your Social Profile"
      id="amSocialLogin.titleText"
    />
  </div>
  <div
    className="undefined undefined undefined"
  >
    <div>
      <button
        className="undefined undefined"
        onClick={[Function]}
        style={
          Object {
            "--amslBtnBg": "var(--amsl-facebook-button__background)",
          }
        }
        title="Sign in with Facebook"
        type="button"
      >
        <span
          className="undefined undefined undefined"
          style={
            Object {
              "--amslBtnBg": "var(--amsl-facebook-button__background)",
              "--amslBtnIcon": "var(--amsl-facebook-icon__background)",
            }
          }
        >
          <span />
        </span>
      </button>
      <button
        className="undefined undefined"
        onClick={[Function]}
        style={
          Object {
            "--amslBtnBg": "var(--amsl-instagram-button__background)",
          }
        }
        title="Sign in with Instagram"
        type="button"
      >
        <span
          className="undefined undefined undefined"
          style={
            Object {
              "--amslBtnBg": "var(--amsl-instagram-button__background)",
              "--amslBtnIcon": "var(--amsl-instagram-icon__background)",
            }
          }
        >
          <span />
        </span>
      </button>
    </div>
  </div>
</div>
`;
